toolchain/check/testdata/function/declaration/no_prelude/name_poisoning.carbon

```diff
@@ -369,6 +369,23 @@ fn F() {
   }
 }
 
+// --- fail_implicitly_poison_core.carbon
+
+// CHECK:STDERR: fail_implicitly_poison_core.carbon:[[@LINE+7]]:9: error: `Core.Bool` implicitly referenced here, but package `Core` not found [CoreNotFound]
+// CHECK:STDERR: fn F(x: bool);
+// CHECK:STDERR:         ^~~~
+// CHECK:STDERR:
+// CHECK:STDERR: fail_implicitly_poison_core.carbon:[[@LINE+3]]:9: error: name used before it was declared [NameUseBeforeDecl]
+// CHECK:STDERR: fn F(x: bool);
+// CHECK:STDERR:         ^~~~
+fn F(x: bool);
+
+// CHECK:STDERR: fail_implicitly_poison_core.carbon:[[@LINE+4]]:1: note: declared here [NameUseBeforeDeclNote]
+// CHECK:STDERR: class Core {}
+// CHECK:STDERR: ^~~~~~~~~~~~
+// CHECK:STDERR:
+class Core {}
```

Comment:
```suggestion
class r#Core {}
```

Reply:
Note, suggesting this specifically so that making `Core` a keyword causes a corresponding behavior change in this test. Which I think is probably fine.

FYI @danakj

Reply:
Done.